File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int l = 0, r = s.size() - 1;
        while(l < s.size() && r > -1) {
            if(tolower(s[l]) == tolower(s[r])) {
                l++;
                r--;
                continue;
            } else if (!isalnum(s[l])) {
                l++;
            } else if (!isalnum(s[r])) {
                r--;
            } else {
                return false;
            }
        }

        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        clean_s = ""
        for i in s:
            if i.isalnum():
                clean_s += i.lower()
        
        l = 0
        r = len(clean_s) - 1
        while l <= r:
            if clean_s[l] != clean_s[r]:
                return False
            l += 1
            r -= 1
        
        return True

